peg_src/jsdoctype.pegjs

```diff
@@ -399,7 +399,10 @@ NumberLiteralExpr = value:(BinNumberLiteralExpr / OctNumberLiteralExpr / HexNumb
                   }
 
 
-DecimalNumberLiteralExpr = $("-"? [0-9]+ ("." [0-9]+)?)
+DecimalNumberLiteralExpr = $(("+" / "-")? UnsignedDecimalNumberLiteralExpr)
+
+UnsignedDecimalNumberLiteralExpr = $([0-9]+ ("." [0-9]+)? ("e" ("+" / "-")? [0-9]+)?)
```

Comment:
This should be its own commit.

Reply:
`UnsignedDecimalNumberLiteralExpr` is needed elsewhere in the commit related to `RecordTypeExprEntryKey` fixes.

I understand not crowding too many things into a PR (like my previously having added the testing refactoring into that other PR), but development will be needlessly slowed down, imo, if we give too much attention to splitting up PRs which are already not too large in content.

I find there is no exact definition for the idea of adding "one thing only" anyways, even if it is a fair guiding principle.

Reply:
If you still want this as a separate PR, I can do so, as I do understand the decimal change spills over into other types (and I've updated the original post to draw more attention to this fact). But it was required in connection with this work.

Reply:
Fine by me.

Reply:
This improves the parsing of numbers, right? Can you add a few tests with the numbers that are now parseable?

Reply:
Good catch. I've added tests. I also liberalized a little further in allowing a decimal without a leading digit and added a test for that.